### ISSUE TITLE
Update DNS extension to use latest rubydns gem

### ISSUE THREAD

**Issue by bcoles** [beefproject/beef#1253]
The DNS extension currently makes use of `rubydns` gem version 0.7 which is also hard-coded in the `Gemfile` and `Gemfile.lock`.

It should be updated to the latest stable version (1.0.3) for maintainability as ongoing work to resolve #1025. This will also help prevent dependency issues like #1249 and #1237 in the future.

[rubydns 0.9.0 onward uses Celluloid instead of EventMachine](https://docs.omniref.com/ruby/gems/rubydns/0.9.0).
- [ ] Remove the hard-coded '~> 0.7.3' version for `rubydns` from `Gemfile`
- [ ] Update the rubydns gem and associated `Gemfile.lock` file with `bundle update rubydns`
- [ ] Verify the `rake unit` and `rake integration` test pass
- [ ] Manually verify the DNS extension continues to work

**Comment by ziflar** [comment]
@bcoles :+1: Thx it's ok 
the update is ok
i just run :
update-beef
beef-xss
it's ok

**Comment by ioquatix** [comment]
RubyDNS 2.0.0-rc1 is available and it should be suitable for your update as it no longer depends on celluloid. Let me know if I can provide any other help.

**Comment by bcoles** [comment]
Thanks @ioquatix

Do you have a migration guide for 0.7 to 2.0 ? In particular, I noticed the 2.0.0 RC1 version gives the following error, even though the gem is `require`d and the documentation indicates that `class Server < RubyDNS::Server` is the correct usage.

```
/pentest/web/beef/extensions/dns/dns.rb:17:in `<module:Dns>': uninitialized constant RubyDNS::Server (NameError)
Did you mean?  TCPServer
```

**Comment by ioquatix** [comment]
I apologise for this, perhaps the documentation is wrong - things have changed a lot - I'll check and get back to you.

**Comment by ioquatix** [comment]
Okay, `RubyDNS::Server` moved to `Async::DNS::Server`. You don't need to depend on `rubydns`, you can just depend directly on `async-dns` and then use `Async::DNS::Server` as your base class.

**Comment by ioquatix** [comment]
RubyDNS 2.0.0 is now released.

**Comment by bcoles** [comment]
Thanks @ioquatix

Unfortunately updating the library is more complicated than I'd hoped and we're short on dev bandwidth.

**Comment by h4sh5** [comment]
Moved the base class to `Async::DNS::Server`, and it starts with `Thread.new { super() }` to not block the program.

**Comment by ioquatix** [comment]
Nice work!

**Comment by bcoles** [comment]
Fixed in f5de5eb7c090023df39256a9983f75fc87403194.
